pkg/utils/utils.go

```diff
@@ -0,0 +1,107 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/logrusorgru/aurora/v4"
+	"github.com/projectdiscovery/pdtm/pkg"
+	"github.com/projectdiscovery/pdtm/pkg/path"
+)
+
+const host = "https://api.pdtm.sh"
+
+// configure aurora for logging
+var au = aurora.New(aurora.WithColors(true))
+
+func FetchToolList() ([]pkg.Tool, error) {
+	tools := make([]pkg.Tool, 0)
+
+	// Get current OS name, architecture, and Go version
+	osName := runtime.GOOS
+	osArch := runtime.GOARCH
+	goVersion := runtime.Version()
+
+	// Create the request URL with query parameters
+	reqURL := host + "/api/v1/tools?os=" + osName + "&arch=" + osArch + "&go_version=" + goVersion
+
+	resp, err := http.Get(reqURL)
```

Comment:
@RamanaReddy0M In callback function we are fetching all tools which don't seem to be necessary since we have a different endpoint for that
```url
https://api.pdtm.sh/api/v1/tools/$projectName
```
I think we can add a new helper function for that using above endpoint